beacon-chain/rpc/eth/helpers/sync.go

```diff
@@ -50,9 +62,23 @@ func IsOptimistic(ctx context.Context, st state.BeaconState, optimisticSyncFetch
 	if err != nil {
 		return false, errors.Wrap(err, "could not get header root")
 	}
-	isOptimistic, err := optimisticSyncFetcher.IsOptimisticForRoot(ctx, headRoot)
+
+	isOptimistic, err := optimisticModeFetcher.IsOptimisticForRoot(ctx, headRoot)
 	if err != nil {
 		return false, errors.Wrap(err, "could not check if block is optimistic")
 	}
+
 	return isOptimistic, nil
 }
+
+type syncDetailsJson struct {
```

Comment:
Duplication added here because I can't confidently decide between:

1) Moving `syncDetailsJson` entirely out of `structs.go` (along with other structs that depend on it...) and importing `sync.go` wherever those structs are used, or
2) Creating a new structs file that could be referenced by both `structs.go` and `sync.go`

My intent is to address that within a separate PR later on unless anyone sees a solution that I can implement within this PR.

Reply:
I think (1) is reasonable